src/model/RacingGame.js

```diff
@@ -0,0 +1,53 @@
+import { OUTPUT_MESSAGE } from '../constant/index.js';
+import { getRandomNumberInRange } from '../utils/random.js';
+import Car from './Car.js';
+
+class RacingGame {
+  #cars;
+
+  #tryNumber;
+
+  constructor(carNames, tryNumber) {
+    this.#cars = carNames.map((carName) => new Car(carName));
+    this.#tryNumber = tryNumber;
+  }
+
+  race() {
+    const raceResult = [];
+    for (let i = 0; i < this.#tryNumber; i += 1) {
+      this.#moveCarsOnce();
+      raceResult.push(this.#serializeCarsState());
+    }
+
+    return raceResult;
+  }
+
+  #moveCarsOnce() {
+    this.#cars.forEach((car) => {
+      const randomNumber = getRandomNumberInRange(0, 9);
+      if (randomNumber >= 4) car.move();
+    });
+  }
+
+  #serializeCarsState() {
+    const serialized = this.#cars.map((car) => {
+      const { name, distance } = car.getCarInformation();
+      const newDistance = Array(distance).fill(OUTPUT_MESSAGE.STRINGYFY_DISTANCE).join('');
+
+      return `${name} : ${newDistance}`;
+    });
+
+    return serialized;
+  }
+
+  getWinners() {
+    const max = Math.max(...this.#cars.map((car) => car.getCarInformation().distance));
+    const winners = this.#cars
+      .filter((car) => car.getCarInformation().distance === max)
+      .map((car) => car.getCarInformation().name);
+
+    return winners;
+  }
+}
```

Comment:
함수를 분리해서 조금 더 선언적으로 작성해주는 것도 좋을 거 같아요!

```suggestion
  #calculateWinnerDistnace() {
    return Math.max(...this.#cars.map((car) => car.getCarInformation().distance));
  }

  getWinners() {
    const winnerDistance = this.#calculateWinnerDistance();
    
    const winners = this.#cars
      .filter((car) => car.getCarInformation().distance === max)
      .map((car) => car.getCarInformation().name);

    return winners;
  }
}
```

winners를 구하는 부분도 비슷하게 바꿀 수 있으면 조금 더 한 눈에 흐름을 파악할 수 있을 거 같아요.

물론 현재도 너무 좋습니다!

Reply:
이 부분에서 더 분리 했어야 했네요...!! 감사합니다 👍👍